File: boxsys/ViewController.m
//
//  ViewController.m
//  boxsys
//
//  Created by chunyi.zhoucy on 14-1-7.
//  Copyright (c) 2014年 chunyi.zhoucy. All rights reserved.
//

#import "ViewController.h"
#import "SBJson/SBJson.h"
#import "AppDelegate.h"

@interface ViewController ()

@end

@implementation ViewController


- (void)viewDidLoad
{
    [super viewDidLoad];
	// Do any additional setup after loading the view, typically from a nib.
    
    NSDictionary *dData = [AppDelegate getDictionaryFromFile:@"JuToday"];
    if (!dData) return;
    [JuBoxSys2 loadBox:@"Today"
               withKey:@"ViewController"
              withData:dData
                 width:320 height:0 withDelegate:self];
    
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

static int modelStatus = 0;

- (IBAction)changeModel:(id)sender
{
    switch (modelStatus) {
        case 0:
            break;
        case 1:
            break;
        case 2:
            break;
        default:
            modelStatus = 0;
            break;
    }
    modelStatus += 1;
    
    // 测试删除对象，看释放情况
    [JuBoxSys2 clearSelf];
}

static int dataStatus = 0;

- (IBAction)changeData:(id)sender
{
    switch (dataStatus) {
        case 0:
            break;
        case 1:
            break;
        default:
            dataStatus = 0;
            break;
    }
    dataStatus += 1;
    
}

//=============================================================================//

- (UIView *) getContainer
{
    if(!self.workview) {
        self.workview = [[[UIView alloc] initWithFrame:CGRectMake(0, 0, 320, 480)] autorelease];
        [self.scrollView addSubview:self.workview];
    }
    return self.workview;
}

- (void) onBoxSysEvent:(id) sender withArgs:(NSDictionary *) argsMap
{
    
}

@end

File: boxsys/ViewController.swift
//
//  ViewController.swift
//  boxsys
//
//  Migrated to Swift from Objective-C
//

import UIKit

class ViewController: UIViewController, JuBoxSysDelegate {

    @IBOutlet weak var scrollView: UIScrollView!

    var workview: UIView?

    private var modelStatus = 0
    private var dataStatus = 0

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.

        guard let dData = AppDelegate.getDictionary(fromFile: "JuToday") else { return }

        JuBoxSys2.loadBox(
            name: "Today",
            key: "ViewController",
            dataMap: dData,
            width: 320,
            height: 0,
            delegate: self
        )
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    @IBAction func changeModel(_ sender: Any) {
        switch modelStatus {
        case 0:
            break
        case 1:
            break
        case 2:
            break
        default:
            modelStatus = 0
        }
        modelStatus += 1

        // 测试删除对象，看释放情况
        JuBoxSys2.clearSelf()
    }

    @IBAction func changeData(_ sender: Any) {
        switch dataStatus {
        case 0:
            break
        case 1:
            break
        default:
            dataStatus = 0
        }
        dataStatus += 1
    }

    // MARK: - JuBoxSysDelegate

    func getContainer() -> UIView {
        if workview == nil {
            workview = UIView(frame: CGRect(x: 0, y: 0, width: 320, height: 480))
            scrollView.addSubview(workview!)
        }
        return workview!
    }

    func onBoxSysEvent(sender: Any, args: [String: Any]?) {
        // Handle box system events
    }
}
